src/compiler/checker.ts

```diff
@@ -18595,20 +18598,23 @@ namespace ts {
                     return true;
                 }
             }
-            else if (targetType.flags & TypeFlags.UnionOrIntersection) {
+            else if (targetType.flags & TypeFlags.UnionOrIntersection && isExcessPropertyCheckTarget(targetType)) {
                 for (const t of (targetType as UnionOrIntersectionType).types) {
                     if (isKnownProperty(t, name, isComparingJsxAttributes)) {
                         return true;
                     }
                 }
             }
-            else if (targetType.flags & TypeFlags.Conditional) {

```

Comment:
![image](https://user-images.githubusercontent.com/972891/49479118-b960f000-f7d6-11e8-8307-7f0d04b0cc21.png)

Reply:
Any assignment of an object literal to a conditional type is an error, so it makes no sense to do excess property checking on the branches of the conditional. The minute you fix errors generated by the excess property checks you'll just get another error saying you can't assign to the conditional type.

Reply:
Makes sense, I thought we just checked if it sources were assignable to both branches